test/integ/quote-classic.test.ts

```diff
@@ -225,6 +227,24 @@ describe('quote', function () {
   for (const algorithm of ['alpha']) {
     for (const type of ['EXACT_INPUT', 'EXACT_OUTPUT']) {
       describe(`${ID_TO_NETWORK_NAME(1)} ${algorithm} ${type} 2xx`, () => {
+        const greenlistCarveout: [Currency, Currency][] = new Array<[Currency, Currency]>();
+        for (let i = 0; i < GREENLIST_USD_STABLES.length; i += 2) {
+          const [tokenIn, tokenOut] = GREENLIST_USD_STABLES.slice(i, i + 2);
+          greenlistCarveout.push([tokenIn, tokenOut]);
+        }
+        for (let i = 0; i < GREENLIST_EUR_STABLES.length; i += 2) {
+          const [tokenIn, tokenOut] = GREENLIST_EUR_STABLES.slice(i, i + 2);
+          greenlistCarveout.push([tokenIn, tokenOut]);
+        }
+
+        const greenlistTokensPairs: [Currency, Currency][] = new Array<[Currency, Currency]>();
+        for (let i = 0; i < GREENLIST_TOKENS.length; i += 2) {
+          const [tokenIn, tokenOut] = GREENLIST_TOKENS.slice(i, i + 2);
+          if (!greenlistCarveout.find((pair) => pair[0].equals(tokenIn) && pair[1].equals(tokenOut))) {
+            greenlistTokensPairs.push([tokenIn, tokenOut]);
+          }
+        }
+
```

Comment:
Is this just getting all possible token pairs within the greenlist? If so this is pretty hard to follow, what is `greenlistCarveout` vs. `greenlistTokensPairs`? What should we expect the end value to be in `greenlistTokensPairs`?

Reply:
Also might want to dedupe arrays or use Sets here depending on implementation